apps/designer/app/designer/features/style-panel/sections/spacing/input-popover.tsx

```diff
@@ -0,0 +1,118 @@
+import { useState } from "react";
+import {
+  keyframes,
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverPortal,
+  styled,
+} from "@webstudio-is/design-system";
+import {
+  CssValueInput,
+  type IntermediateStyleValue,
+} from "../../shared/css-value-input";
+import type { StyleProperty, StyleValue } from "@webstudio-is/css-data";
+import type { StyleChangeEvent } from "./types";
+
+const slideUpAndFade = keyframes({
+  "0%": { opacity: 0, transform: "scale(0)" },
+  "100%": { opacity: 1, transform: "scale(1)" },
+});
+
+const Input = ({
+  value,
+  property,
+  onChange,
+  onClosePopover,
+}: {
+  property: StyleProperty;
+  value: StyleValue;
+  onChange: (event: StyleChangeEvent) => void;
+  onClosePopover: () => void;
+}) => {
+  const [intermediateValue, setIntermediateValue] = useState<
+    StyleValue | IntermediateStyleValue
+  >();
+
+  return (
+    <CssValueInput
+      property={property}
+      value={value}
+      intermediateValue={intermediateValue}
+      onChange={(styleValue) => {
+        setIntermediateValue(styleValue);
+        if (styleValue.type !== "intermediate") {
+          onChange({ property, value: styleValue, isEphemeral: true });
+        }
+      }}
+      onHighlight={(styleValue) => {
+        onChange({ property, value: styleValue, isEphemeral: true });
+      }}
+      onChangeComplete={(styleValue, reason) => {
+        setIntermediateValue(undefined);
+        onChange({ property, value: styleValue, isEphemeral: false });
+
+        // @todo: handle Esc
+        if (reason === "blur" || reason === "enter") {
+          onClosePopover();
+        }
+      }}
+    />
+  );
+};
+
+// trigger is used only for positioning
+const Trigger = styled("div", { position: "absolute", width: 0, height: 0 });
+
+const PopoverContentStyled = styled(PopoverContent, {
+  minWidth: 0,
+  minHeight: 0,
+  width: 88,
+  border: "1px solid $colors$slate8",
+  borderRadius: "$borderRadius$7",
+  background: "$colors$gray2",
+  padding: "$spacing$5",
+  boxShadow: "0px 2px 7px rgba(0, 0, 0, 0.1), 0px 5px 17px rgba(0, 0, 0, 0.15)",
```

Comment:
Usually most box shadows are same, and should be in tokens too?

Reply:
We currently don't have boxShadows in `stitches.config.ts`, all of them defined in code like here. 

In Figma it also seem to be unique:
<img width="539" alt="Screenshot 2022-12-21 at 11 31 20" src="https://user-images.githubusercontent.com/825702/208846768-f8665e3a-368d-4546-8c3e-63f405696d78.png">

Maybe we should start defining them in `stitches`, but not sure I should do it in this PR.

Reply:
we are going to hav tokens for those too, defined in figma tokens plugin and synced into the codebase, I guess for now we could do either way, might be easier to migrate if its in stitches config to see them all at once